src/cnc/flavors/aws/lambda-lite/1/deploy/deploy_functions.sh.j2

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+# Assigning variables
+name_lambda="{{ service.instance_name }}"
+output_dir="{{ service.environment.collection.config_files_path }}/build/{{ service.instance_name }}_lambda_function.zip"
+region="{{ service.environment.collection.region }}"
+
+# Validation echos
+echo "Validating variables for service {{ service.instance_name }}..."
+echo "name_lambda: $name_lambda"
+echo "output_dir: $output_dir"
+echo "region: $region"
+
+
+env_vars_file="env_vars.json"
+echo "Creating environment variables file $env_vars_file..."
+
+# Start JSON file
+echo '{ "Variables": {' > $env_vars_file
+
+{% for item in service.insecure_environment_items %}
+echo "  \"{{item.name}}\": \"{{item.value}}\"," >> $env_vars_file
+{% endfor %}
+sed -i '$ s/,$//' $env_vars_file
+
+# Close JSON file
+echo '}}' >> $env_vars_file
+
+cat $env_vars_file
+
+# Check if variables are set
+if [ -z "$name_lambda" ] || [ -z "$output_dir" ] || [ -z "$region" ]; then
+  echo "Error: One or more required variables are not set for service {{ service.instance_name }}."
+  exit 1
+fi
+
+# Deploying lambda function using AWS CLI
+echo "Deploying lambda function $name_lambda with ZIP file at $output_dir in region $region"
+
+# Lambda deployment command using AWS CLI
+aws lambda update-function-code --function-name "$name_lambda" --zip-file "fileb://$output_dir" --region "$region" --no-cli-pager
+
+# Capturing the exit status of the aws command
+exit_code=$?
+
+# Checking the exit status of the aws command
+if [ $exit_code -eq 0 ]; then
+  echo "Lambda function $name_lambda updated successfully"
+else
+  echo "Error updating lambda function $name_lambda."
+  exit $exit_code
+fi
+
+# Updating the lambda function configuration to add environment variables
+echo "Updating environment variables for lambda function $name_lambda"
+aws lambda update-function-configuration --function-name "$name_lambda" --environment "file://$env_vars_file" --region "$region" --no-cli-pager
```

Comment:
Just want to raise a question here: What happens if the function update succeeds but the env var update does not? We may eventually need some kind of rollback logic for the function

Reply:
Ajustments Commit: https://github.com/tmaior/cnc-fork/commit/9ed6e46909e2e43c43eee39bb138ed6d2df70b18